diff --git a/SourceFiles/UtilityFunctions.cpp b/SourceFiles/UtilityFunctions.cpp
index 33e543f..9466ffd 100644
--- a/SourceFiles/UtilityFunctions.cpp
+++ b/SourceFiles/UtilityFunctions.cpp
@@ -125,7 +125,7 @@ void run_cpuid(uint32_t eax, uint32_t ecx, int32_t *abcd)
 	abcd[3] = d;
 }
 
-static uint64_t _xgetbv(uint32_t xcr)
+static uint64_t _xgetbv_fixed(uint32_t xcr)
 {
 	uint32_t eax, edx;
 
@@ -138,7 +138,11 @@ static uint64_t _xgetbv(uint32_t xcr)
 int32_t check_xcr0_ymm()
 {
 	uint32_t xcr0;
-	xcr0 = (uint32_t)_xgetbv(0);
+#if defined(__GNUC__) || defined(__clang__)
+    xcr0 = (uint32_t)_xgetbv_fixed(0);
+#else
+    xcr0 = (uint32_t)_xgetbv(0);
+#endif
 	return ((xcr0 & 6) == 6); /* checking if xmm and ymm state are enabled in XCR0 */
 }
 
